src/xrGame/configs_dump_verifyer.cpp

```diff
@@ -197,20 +197,17 @@ bool const configs_verifyer::verify(u8* data, u32 data_size, string256& diff)
 
     m_orig_config_body.w_stringZ(add_str);
 
-    crypto::xr_sha256 tmp_sha_checksum;
-    tmp_sha_checksum.start_calculate(m_orig_config_body.pointer(), m_orig_config_body.tell());
-    while (!tmp_sha_checksum.continue_calculate())
-    {
-    };
+    crypto::xr_sha1::hash_t hash{};
+    crypto::xr_sha1::calculate(hash, m_orig_config_body.pointer(), m_orig_config_body.tell());
 
-    u8 tmp_checksum[crypto::xr_sha256::digest_length];
+    crypto::xr_sha1::hash_t tmp_checksum{};
```

Comment:
Zero-initializing of `tmp_checksum` is unnecessary here because its value is used only when `verify_dsign()` returns `true`.

Reply:
hash_t is an std::array which is a class type. **It will be default-initialized whether with curved brackets or not.**

Reply:
What's the point of having brackets then? If it will be initialized in any way, then there's no need to have brackets.
```suggestion
    crypto::xr_sha1::hash_t tmp_checksum;
```